pygmt/clib/loading.py

```diff
@@ -96,20 +97,27 @@ def clib_names(os_name: str) -> list[str]:
     return libnames
 
 
-def clib_full_names(env=None):
+def clib_full_names(env: os._Environ | dict | None = None) -> Iterator[str]:
```

Comment:
Not sure about using a private `os_Environ` here...

Reply:
`os.environ` returns a mapping object, so change it to `collections.abc.Mapping` in 3212ac94f3f79897a11cd90ff97a06323c752447